src/Tribe/Commerce/PayPal/Main.php

```diff
@@ -870,101 +866,59 @@ public function generate_tickets( $payment_status = 'completed', $redirect = tru
-					update_post_meta( $attendee_id, $this->security_code, $this->generate_security_code( $attendee_id ) );
-					update_post_meta( $attendee_id, $this->order_key, $order_id );
-					update_post_meta( $attendee_id, $this->attendee_optout_key, $optout );
-					update_post_meta( $attendee_id, $this->email, $individual_attendee_email );
-					update_post_meta( $attendee_id, $this->full_name, $individual_attendee_name );
-					update_post_meta( $attendee_id, '_paid_price', get_post_meta( $product_id, '_price', true ) );
-					update_post_meta( $attendee_id, '_price_currency_symbol', $currency_symbol );
-				}
+				$data = $attendee;
 
-				update_post_meta( $attendee_id, $this->attendee_tpp_key, $attendee_order_status );
+				$data['order_attendee_id'] = $order_attendee_id;
+				$data['attendee_status']   = $attendee_order_status;
 
 				if ( Tribe__Tickets__Commerce__PayPal__Stati::$refunded === $payment_status ) {
 					$refund_order_id = Tribe__Utils__Array::get( $transaction_data, 'txn_id', '' );
-					update_post_meta( $attendee_id, $this->refund_order_key, $refund_order_id );
+
+					$data['refund_order_id'] = $refund_order_id;
 				}
 
 				if ( ! $updating_attendee ) {
-					/**
-					 * Action fired when an PayPal attendee ticket is created
-					 *
-					 * @since 4.7
-					 *
-					 * @param int    $attendee_id           Attendee post ID
-					 * @param string $order_id              PayPal Order ID
-					 * @param int    $product_id            PayPal ticket post ID
-					 * @param int    $order_attendee_id     Attendee number in submitted order
-					 * @param string $attendee_order_status The order status for the attendee.
-					 */
-					do_action( 'event_tickets_tpp_attendee_created', $attendee_id, $order_id, $product_id, $order_attendee_id, $attendee_order_status );
-				}
+					$optout = Tribe__Utils__Array::get( $attendee_optouts, 'ticket_' . $product_id, false );
+					$optout = filter_var( $optout, FILTER_VALIDATE_BOOLEAN );
+					$optout = $optout ? 'yes' : 'no';
 
-				/**
-				 * Action fired when an PayPal attendee ticket is updated.
-				 *
-				 * This action will fire both when the attendee is created and
-				 * when the attendee is updated.
-				 * Hook into the `event_tickets_tpp_attendee_created` action to
-				 * only act on the attendee creation.
-				 *
-				 * @since 4.7
-				 *
-				 * @param int    $attendee_id           Attendee post ID
-				 * @param string $order_id              PayPal Order ID
-				 * @param int    $product_id            PayPal ticket post ID
-				 * @param int    $order_attendee_id     Attendee number in submitted order
-				 * @param string $attendee_order_status The order status for the attendee.
-				 */
-				do_action( 'event_tickets_tpp_attendee_updated', $attendee_id, $order_id, $product_id, $order_attendee_id, $attendee_order_status );
+					$data['ticket_id']       = $product_id;
+					$data['post_id']         = $post_id;
+					$data['order_id']        = $order_id;
```

Comment:
:warning: Equals sign not aligned with surrounding assignments; expected 7 spaces but found 8 spaces

<sub>hash: a4d5ed25c1aee17555464b63fe0c4c8a</sub>